### PR TITLE
Add equals and hashCode for Content.java

### PR DESCRIPTION
**Pull request by way-zer**
If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

When content like Blocks.arc reassign to new instance, references in containers will be wrong.

### CODE REVIEW

File: core/src/mindustry/ctype/Content.java [hunk 1]
```diff
@@ -49,6 +51,18 @@ public String toString(){
...
+    @Override
+    public boolean equals(Object o){
+        if(this == o) return true;
+        if(!(o instanceof Content content)) return false;
+        return id == content.id && getContentType() == content.getContentType();
+    }
+
+    @Override
+    public int hashCode(){
+        return Objects.hash(getContentType(), id);
```
Comment (Anuken) [inline core/src/mindustry/ctype/Content.java:63]: This is unsupported Java 8 API.

**Comment by Anuken** [comment]
> When content like Blocks.arc reassign to new instance, references in containers will be wrong.

This sounds like a really bad idea in general.

**Comment by way-zer** [comment]
> > When content like Blocks.arc reassign to new instance, references in containers will be wrong.
> 
> This sounds like a really bad idea in general.
> 
> Edit: It would also be misleading - you should use (possibly packed) `id | type << 16` keys, not objects. Your modified arc instance would not actually be equal in fields, it would just have the same ID.

As content type always single instance, the `equal` not bad. Just make new instance still be the same one as the old.

**Comment by way-zer** [comment]
Wait, one problem, if mods set old instance to block. Need to add a helper function like `reGet`.

**Comment by way-zer** [comment]
I try to resolve this in plugin internally,

**Pull request closed without merging**